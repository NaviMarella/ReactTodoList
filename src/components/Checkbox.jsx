import PropTypes from 'prop-types';
import React, { Component } from 'react';
import "./Checkbox.css";

class Checkbox extends Component {
  toggleCheckboxChange = () => {
    const { handleCheckboxChange } = this.props;
    handleCheckboxChange();
  }

  render() {
    const { label, isChecked } = this.props;

    return (
      <div className="checkbox_container" >
        <label className="container">
            {label}
            <input type="checkbox" onChange={this.toggleCheckboxChange} checked={isChecked}  />
            <span className="checkmark"></span>
        </label>
      </div>
    );
  }
}

Checkbox.defaultProps = {
  type: "checkbox"
}

Checkbox.propTypes = {
  label: PropTypes.string,
  isChecked: PropTypes.bool,
  handleCheckboxChange: PropTypes.func.isRequired
};


export default Checkbox;