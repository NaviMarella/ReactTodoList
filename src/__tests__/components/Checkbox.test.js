import { shallow } from 'enzyme';
import React from 'react';
import Checkbox from './../../components/Checkbox';

describe('Checkbox Component', () => {
  test('render snapshot', () => {
      let props = {
        isChecked: true,
        handleCheckboxChange: jest.fn()
      }
    const wrapper = shallow(<Checkbox {...props} />);
     wrapper.find('input[type="checkbox"]').simulate('change',{ target: { checked: true } });
     expect(props.handleCheckboxChange).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();

  });
});
