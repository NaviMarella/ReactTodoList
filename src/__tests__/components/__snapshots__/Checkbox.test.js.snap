// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component render snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox
    handleCheckboxChange={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    isChecked={true}
    type="checkbox"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <label
        className="container"
      >
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="checkmark"
        />
      </label>,
      "className": "checkbox_container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <input
            checked={true}
            onChange={[Function]}
            type="checkbox"
          />,
          <span
            className="checkmark"
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": true,
            "onChange": [Function],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "checkmark",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "label",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <label
          className="container"
        >
          <input
            checked={true}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="checkmark"
          />
        </label>,
        "className": "checkbox_container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <input
              checked={true}
              onChange={[Function]}
              type="checkbox"
            />,
            <span
              className="checkmark"
            />,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": true,
              "onChange": [Function],
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "checkmark",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "label",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
